### PR TITLE
Package brotli.1.3.0

### PR DESCRIPTION
**Pull request by fxfactorial**
### `brotli.1.3.0`

Bindings to Google's Brotli compresion algorithm

OCaml Bindings to brotli, Google's compression algorithm for the web
Source: https://github.com/google/brotli/ RFC:
http://www.ietf.org/id/draft-alakuijala-brotli

---
* Homepage: http://hyegar.com
* Source repo: https://github.com/fxfactorial/ocaml-brotli.git
* Bug tracker: https://github.com/fxfactorial/ocaml-brotli/issues

---

:camel: Pull-request generated by opam-publish v0.3.5

### CODE REVIEW

**Comment by fxfactorial** [comment]
Installation on Linux expected to fail as required external library, brotli so far only is on Debian Sid, not in Jessie yet and I don't know how or if it is even okay to target Sid (Is that possible to target Sid or Jessie) in an opam file.

**Comment by dbuenzli** [comment]
@fxfactorial best would be to add a `conf-brotli` package with appropriate `depexts` as I suggested in #5882 without this it's unlikely that any of the CI will pass and it will greatly improve the experience of the users of your package.

**Comment by fxfactorial** [comment]
@dbuenzli you're right! I agree, right now it only has depexts for OS X, but Linux being more important for this code. I haven't written one of these `conf` packages yet, so I need to get around to that. What is a simple example?

**Comment by dbuenzli** [comment]
[Here's](https://github.com/ocaml/opam-repository/tree/master/packages/conf-sdl2/conf-sdl2.1) one for a C library.

**Comment by fxfactorial** [comment]
@dbuenzli but the one I need is this: 

https://packages.debian.org/search?keywords=libbrotli0.6.0

that's what this code was built against. I don't see a way in depext to say experimental?

**Comment by dbuenzli** [comment]
> that's what this code was built against. I don't see a way in depext to say experimental?

Don't worry just add the info as it will eventually be. The `depext` mecanism is currently under powered and being reworked, so far it cannot express these kind of things. When you PR the `opam-repository` just let the mergers know why the failure occurs, see for example [here](https://github.com/ocaml/opam-repository/pull/9914#issuecomment-317581361).

**Comment by fxfactorial** [comment]
@dbuenzli  Okay, will do that now. Why do you say to make a separate conf package rather than just adding the depexts in the main package's opam file directly? as I have now: 

```
depexts: [
  [["osx" "homebrew"] ["brotli"]]
]
```

Are we now doing dedicated separate conf packages for depexts?

**Comment by dbuenzli** [comment]
> Why do you say to make a separate conf package rather than just adding the depexts in the main package's opam file directly?

First, other packages using the C library can then use this `conf-brotli` packages. Second it's not DRY to do so: every of your release will have the set of depexts, now if you need to alter/add one we need to modify all the `opam` files of your previous releases.

File: packages/brotli/brotli.1.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"]
+  ["ocaml" "setup.ml" "-build"]
+  ["ocaml" "setup.ml" "-test"]
+]
+build-doc: ["ocaml" "setup.ml" "-doc"]
+remove: ["ocamlfind" "remove" "brotli"]
+depends: [
+  "oasis" {build}
+  "ocamlfind" {build}
+  "ocamlbuild" {build}
```
Comment (dbuenzli) [inline packages/brotli/brotli.1.3.0/opam]: You'll need to add `conf-brotli` here.

Comment (fxfactorial) [inline packages/brotli/brotli.1.3.0/opam]: Fixed and just opened the conf-brotli package.

Comment (dbuenzli) [inline packages/brotli/brotli.1.3.0/opam]: Excellent thanks. We'll restart the CI once the `conf-brotli` package is merged.

**Comment by dbuenzli** [comment]
Now that #10120, the CI should be restarted.

**Comment by avsm** [comment]
Need to also merge this PR branch against the `master` branch, and then CI will pick up the package

**Comment by dbuenzli** [comment]
@fxfactorial could you please rebase your PR.

**Comment by fxfactorial** [comment]
rebased

**Comment by dbuenzli** [comment]
The linux failures are [expected](https://github.com/ocaml/opam-repository/pull/10118#issuecomment-323585730), travis macOS is fine. This is good to go.

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Pull request merged**